snap/snapcraft.yaml

```diff
@@ -1,52 +1,127 @@
   pyspark:
     command: bin/pyspark
     environment:
       PYTHONPATH: $PYTHONPATH:$SNAP/usr/lib/python3/dist-packages
+    plugs:
+        - network
+        - network-bind
+        - home
 
 parts:
   spark:
-    source: https://github.com/apache/spark.git
     plugin: nil
-    build-packages: 
-        - git
-        - openjdk-11-jdk
-        - maven
+    build-packages:
+        - ca-certificates
+        - ca-certificates-java
+        - openjdk-11-jre-headless
         - python3
-        - python3-pip
         - wget
     stage-packages:
-        - openjdk-11-jre
+        - openjdk-11-jre-headless
     override-build: |
-        dev/make-distribution.sh -Pkubernetes
-        cd dist/jars
-        wget https://repo1.maven.org/maven2/net/java/dev/jets3t/jets3t/0.9.4/jets3t-0.9.4.jar
-        wget https://repo1.maven.org/maven2/com/amazonaws/aws-java-sdk-bundle/1.11.874/aws-java-sdk-bundle-1.11.874.jar
-        wget https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-aws/3.2.2/hadoop-aws-3.2.2.jar
-        cd ../..
+        SPARK_VERSION=$(curl --silent https://downloads.apache.org/spark/ | grep "spark-" | cut -d'>' -f3 | cut -d'/' -f1  | sort | tail -n 1)
+        STATUSCODE=$(curl --silent --head https://downloads.apache.org/spark/${SPARK_VERSION}/${SPARK_VERSION}-bin-hadoop3.tgz | head -n 1 | cut -d' ' -f2)
+        echo "====================================="
+        echo ~${SPARK_VERSION}~
+        echo ~${STATUSCODE}~
+        echo "====================================="
+        if  [[ ${STATUSCODE} -ne 200 ]]
+          then
+            echo "ERROR: Latest available Spark version ${SPARK_VERSION} does not have a downloadable binary! Exiting...."
+            exit 1
+        fi
+        echo "Downloading latest available Spark version ${SPARK_VERSION}."
+        SPARK_HADOOP_VERSION='3'
+        AWS_JAVA_SDK_BUNDLE_VERSION='1.11.874'
+        HADOOP_AWS_VERSION='3.2.2'
+        wget https://downloads.apache.org/spark/${SPARK_VERSION}/${SPARK_VERSION}-bin-hadoop${SPARK_HADOOP_VERSION}.tgz
+        tar -zxf ${SPARK_VERSION}-bin-hadoop${SPARK_HADOOP_VERSION}.tgz
+        cd ${SPARK_VERSION}-bin-hadoop${SPARK_HADOOP_VERSION}/jars
+        wget https://repo1.maven.org/maven2/com/amazonaws/aws-java-sdk-bundle/${AWS_JAVA_SDK_BUNDLE_VERSION}/aws-java-sdk-bundle-${AWS_JAVA_SDK_BUNDLE_VERSION}.jar
+        wget https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-aws/${HADOOP_AWS_VERSION}/hadoop-aws-${HADOOP_AWS_VERSION}.jar
+        cd ..
         mkdir -p $SNAPCRAFT_PRIME/bin
-        cp -r dist/bin/* $SNAPCRAFT_PRIME/bin/
+        cp -r bin/* $SNAPCRAFT_PRIME/bin/
         mkdir -p $SNAPCRAFT_PRIME/jars
-        cp -r dist/jars/* $SNAPCRAFT_PRIME/jars/
+        cp -r jars/* $SNAPCRAFT_PRIME/jars/
+        mkdir -p $SNAPCRAFT_PRIME/python
+        cp -r python/* $SNAPCRAFT_PRIME/python/
+    override-prime: |
+        snapcraftctl prime
+        rm -vf usr/lib/jvm/java-11-openjdk-*/lib/security/blacklisted.certs
```

Comment:
Is any reason why we delete blacklisted certs?

Reply:
Snap build fails if we do not do this. Snappy debug revealed this.